File: app/src/main/java/com/ladsoft/bakingapp/data/database/translator/IngredientTranslator.java
package com.ladsoft.bakingapp.data.database.translator;

import com.ladsoft.bakingapp.data.database.entity.IngredientRecord;
import com.ladsoft.bakingapp.entity.Ingredient;
import com.ladsoft.bakingapp.translator.Translator;

public class IngredientTranslator implements Translator<IngredientRecord, Ingredient>{
    @Override
    public Ingredient translate(IngredientRecord source) {
        return new Ingredient(source.getId(),
                            source.getRecipeId(),
                            source.getQuantity(),
                            source.getMeasure() != null ? source.getMeasure() : "",
                            source.getDescription() != null ? source.getDescription() : "");
    }
}

File: app/src/main/java/com/ladsoft/bakingapp/data/database/translator/IngredientTranslator.kt
package com.ladsoft.bakingapp.data.database.translator

import com.ladsoft.bakingapp.data.database.entity.IngredientRecord
import com.ladsoft.bakingapp.entity.Ingredient
import com.ladsoft.bakingapp.translator.Translator

class IngredientTranslator: Translator<IngredientRecord, Ingredient> {
    override fun translate(source: IngredientRecord): Ingredient {
        return Ingredient(source.id, source.recipeId, source.quantity, source.measure, source.description)
    }
}